entsoe/parsers.py

```diff
@@ -346,15 +433,29 @@ def _parse_datetimeindex(soup):
     Parameters
     ----------
     soup : bs4.element.tag
+    tz: str
 
     Returns
     -------
     pd.DatetimeIndex
     """
     start = pd.Timestamp(soup.find('start').text)
     end = pd.Timestamp(soup.find('end').text)
+    if not tz is None:
+        start = start.tz_convert(tz)
+        end = end.tz_convert(tz)
+
     delta = _resolution_to_timedelta(res_text=soup.find('resolution').text)
     index = pd.date_range(start=start, end=end, freq=delta, closed='left')
+    if not tz is None:
+        dst_jump = len(set(index.map(lambda d:d.dst()))) > 1
+        if dst_jump and delta == "7D":
+            # For a weekly granularity, if we jump over the DST date in October,
+            # date_range erronously returns an additional index element
+            # because that week contains 169 hours instead of 168.
+            index = index[:-1]
```

Comment:
Can you provide an example of a query where this is applied? Because I currently don't see how omitting the last index yields a correct index...
How does this behave in March, when the DST jump adds an extra hour?

Reply:
This one is tricky: remarkably the weekly responses only create an issue for the October DST transition, never in March.